cumulusci/robotframework/tests/salesforce/api.robot

```diff
@@ -101,22 +101,13 @@ Collection API Errors Test
         ...  FirstName=User {{number}}
         ...  LastName={{fake.last_name}}
         ...  Xyzzy=qwertz
-    Run Keyword And Expect Error   SalesforceMalformedRequest*   Salesforce Collection Insert  ${objects}
+    Run Keyword And Expect Error   *Xyzzy*   Salesforce Collection Insert  ${objects}
```

Comment:
This modification seems rather imprecise. It's essentially saying we don't care what error is thrown as long as it mentions the bad field. We could instead modify it to do accept a regexp , unless you think this is suffienct

For example, this syntax should work if we care to check for either the old or new error:

```
     Run Keyword And Expect Error   REGEXP: ^(SalesforceMalformedRequest|Error on Object 0):*
     ...  Salesforce Collection Insert  ${objects}
```

Reply:
The reason the test broke in the first place was because the regexp was too precise and I'm trying not to repeat that mistake. What value does it offer to prove that the error has the syntax that I expect? The could change the syntax again in the next version. It's just text designed for a human being to read...

Reply:
@prescod "SalesforceMalformedRequest" is an exception class in simple-salesforce, not text from the platform. So I'm still not sure I understand what was the change that precipitated this: did the platform start returning a different response status such that simple-salesforce raises a different exception class, or did simple-salesforce change how it reports exceptions?

Reply:
Yes, the platform went from returning 400, which SimpleSalesforce interprets as an error/exception to returning 200.

It turns out that on deeper investigation, the platform does return the error in a different way in the 200 result, so I've added code to look for it.